hudi-client/hudi-java-client/pom.xml

```diff
@@ -69,6 +69,14 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hudi</groupId>
+            <artifactId>hudi-io</artifactId>
+            <version>${project.version}</version>
+            <classifier>tests</classifier>
+            <type>test-jar</type>
```

Comment:
Is `<type>test-jar</type>` needed?

Reply:
will come back to this when all other comments are addressed